frontend/components/data-table.js

```diff
@@ -0,0 +1,550 @@
+          >
+        </td>
+        
+        ${Object.entries(config).map(([key, value]) => ` 
+        ${value.hasPhoto ? `
+        <td 
+          data-element="detail-container"
+        >
+          <img src="${phone[key]}">
+        </td>` : `
+        <td 
+          data-element="detail-container"
+          data-type="${key}"
+          ${value.isSearchable ? 'data-searchable' : ''}
+          ${value.isEditable ? 'data-edit="false"' : ''}
+        >
+          ${phone[key]}
+        </td>`}
+        `).join('')}
+      </tr>`;
+  }
+
+  _renderPhones() {
+    const { phones } = this._props;
+
+    this._table.insertAdjacentHTML(
+      'beforeend',
+      `
+      ${phones
+    .map(phone => this._generatePhonesListHTML(phone)).join('')}
+    `,
+    );
+  }
+
+  _renderTableTitle() {
+    const { config } = this._props;
+    const { mainCheckboxValue } = this._state;
+
+    this._table.innerHTML = `
+      <tr>
+        <th>
+          <input
+            data-element="phones-checkbox"
+            type="checkbox"
+            ${mainCheckboxValue ? 'checked' : ''}
+          >
+        </th>
+        
+          ${Object.entries(config).map(([key, value]) => `
+          <th 
+              ${value.isSortable ? `data-sortable-key=${key}` : ''}
+          >
+            ${value.title || ''}
+          </th>
+            `).join('')}
+      </tr>`;
+  }
+
+  _renderTableContent() {
+    this._table = this._getElement('phones-block');
+
+    this._renderTableTitle();
+    this._renderPhones();
+  }
+
+  _updateView() {
+    this._renderTableContent();
+  }
+
+  _render() {
+    this._element.innerHTML = `
+      <header class="header">
+        <div class="container">
+          <div class="header__content">
+            <div class="header__logo">Phones.ua</div>
+            <div data-component="pagination-selector"></div>
+            <div class="header__find-input-wrapper">
+              <input
+               data-element="find-input"
+               type="text"
+               class="header__find-input"
+               placeholder="Find something..."/>
+            </div>
+            <div class="header-sort-buttons">
+              <button
+               data-type="filter-selected"
+               data-value="all"
+               class="button header-button-all header-button-active"
+               >
+                 Show All
+               </button>
+              <button
+               data-type="filter-selected"
+               data-value="checked"
+               class="button header-button-selected"
+               >
+                 Show Selected
+               </button>
+            </div>
+          </div>
+        </div>
+        </header>
+        <main class="main">
+        <div class="container">
+          <table data-element="phones-block" class="main__table">
+          </table>
+          <div data-component="pagination"></div>
+        </div>
+      </main>
+      `;
+  }
+}
```

Comment:
don't have more time for review now
but the file appeared to be huge
it a point to think how to split it